### PR TITLE
Remove firebase

### LINKED ISSUE #7438: No Push Notifications are received when app is closed (on IOS)
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug
*Summary*: introduced by https://github.com/status-im/status-react/pull/7268. Basically it means if app is closed, user can't receive PN on IOS.
Issue is created because it worked in another way on 0.9.32, so it is a placeholder for possible discussion around expected behavior.
From PR (https://github.com/status-im/status-react/pull/6893#issuecomment-453983432):

> This is only expected on iOS. Android has provisions to deal with starting up the app through a background task if needed (see link to docs in OP). On iOS, we should show the notifications to the user once he starts the app/logs back in.

#### Expected behavior
user can see PN for incoming message

#### Actual behavior
no PN

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

- User1: Open Status
- User1: add User2 to contact list
- User2: add User1 to contact list
- User1: close app
- User2: send message to 1-1 chat

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: RC 0.9.33
* Operating System: IOS

### LINKED ISSUE #8216: Error: There is no route defined for key home when log in after tapping push notification while status is closed (iOS 12)
# Problem

Status is crashing when i log in after I tap push notification while status is not running on background. it happens on iOS 12. 

## Additional info

- build: nightly / release 0.12.2
- OS: iOS 12 and higher
- device: reproduced on iPhone XR

## Steps

1. open status on iOS
2. add a contact (device B)
3. send status to background
4. send a message to device A from Device B
5. receive notification (do not touch it)
6. kill status from background so app is not running
7. tap the push notification
8. enter password and login

image

```
Error: There is no route defined for key home. Must be one for: 'login','progress','create-account','recover','accounts'

This error is located at:

in h
in _
in N
in s
in c
in h
in N
in RCTView
in reagent2
in root
in reagent1
in RCTView
in RCTView
in c

```

### LINKED ISSUE #8217: Empty mailserver added after app opened from PN with account autologin enabled [Android]
*Type*: Bug

*Summary*: When contact opens an app from PN with autologin enabled - in the list of mailservers there is an empty-value mailserver is added.
If to switch to this <no-value> mailserver then simultaneously 'Autoselection mailserver' setting is enabled -> connection to the <no-value> mailserver fails and contact switches to valid mailserver as expected.

With no autologin it's all fine (no empty mailserver in the list)
If open the app not from PN everything is OK as well  

No issues on iOS, but confirmed with Android

#### Expected behavior

No unexpected mailserver values are added in the list after app opened from PN with account autologin enabled 

#### Actual behavior

Empty mailserver added after app opened from PN with account autologin enabled [Android]
Screenshot 2019-05-15 at 17 36 24

### Reproduction

- Open Status and create new account (`User1`)
- `User1` re-logins with 'Autologin' setting enabled
- `User1` receives contact request from `User2` and adds `User2` in contacts
- `User1` closes Status app
- `User2` sends a message to `User1` (`User1` should receive Push Notification after that)
- `User1` taps Push Notification
- `User1` navigated to `Profile` -> `Mailserver settings`

### Additional Information

- Status version: Develop 0.13.0, Release 0.12.1, Release 0.12.2
- Operating System: Android

#### Logs
Logcat from `User1` above when user opened app from PN
(note: errors after `05-15 17:32:43.304  1495  1529 E ReactNativeJS: ERROR [status-im.mailserver.core:113] - mailserver: add-peer error invalid URL scheme, want "enode"
` are loged after I selected empty mailserver)
[NoValueMailserverFromPN.log](https://github.com/status-im/status-react/files/3183017/NoValueMailserverFromPN.log)

### LINKED ISSUE #8550: Never ending loading indicator is shown on white screen if open app from push notification
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)

On Android: white screen with never ending loading is shown if open app from push notification. Full logcat is also attached but the cause might be this error:
```
07-10 11:27:24.792 14445 14488 E ReactNativeJS: ERROR [status-im.utils.fx:31] - Merging fx with common-keys:  #{:db :status-im.ui.screens.navigation/navigate-to}
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: console.error: "ERROR [status-im.utils.fx:31] - Merging fx with common-keys:  #{:db :status-im.ui.screens.navigation/navigate-to}", stack:
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: hca@11:86988
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: a@11:89969
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: e@11:1958201
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: l@11:1958046
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:702880
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:706641
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: Ib@11:145660
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: n@11:27054
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: pe@11:67422
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: MO@11:705938
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: vBb@11:724761
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:725275
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: md@11:47268
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: ob@11:49822
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:26829
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:68164
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: o@11:725207
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: wrc@11:2052700
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:2054547
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:725290
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: md@11:47268
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: ob@11:49822
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:26829
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:68164
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: o@11:725207
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: a@11:2054242
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:2054116
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:725290
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: md@11:47268
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: ob@11:49822
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:26829
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:68164
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: o@11:725207
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:2242120
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:664157
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: cwb@11:656014
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: gwb@11:657164
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: jwb@11:659247
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: iwb@11:657487
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:658477
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: QN@11:659114
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: n@11:657551
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: y@80:657
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: C@80:1021
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: callImmediates@80:3216
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: callImmediates@-1
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@19:2873
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@19:1264
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@19:2565
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@19:1234
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@-1
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@-1
```

Screenshot_20190710-112955

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)

#### Expected behavior

[comment]: # (Describe what you expected to happen.)
Chat with a message from contact is opened
#### Actual behavior

[comment]: # (Describe what actually happened.)
White screen with loading indicator is shown for at least 5 minutes
### Reproduction

[comment]: # (Describe how we can replicate the bug step by step.)

- two users add each other as contacts an send some text msg in 1:1 chat
- user1 on Android closes the app
- user2 sends a text msg in 1;1 chat
- user1 gets push notification and taps on it.  Check if user1 ends up in the 1:1 chat with user2

### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version: nightly from July 10
- Operating System: Android 9 (real pixel XL)

#### Logs

[comment]: # (if on Android please replicate bug whilst running adb logcat)

[not2.log](https://github.com/status-im/status-react/files/3376806/not2.log)

### LINKED ISSUE #8554: [Maybe Later] for notifications setting in onboarding does not disable notifications
# Problem

Tapping `Maybe later` does not make effect, so notification setting will be enabled no matter which option i select. This applies only for Android devices.

image

## Implementation

Tapping `Maybe later` does not enable notifications. Notifications settings are off for Status app in device settings.

### LINKED ISSUE #8555: no push notification shown if send message in 1:1 between 0.13.0 release and nightly
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: no pn is shown if send message from 0.13.0 in 1:1 chat to user with nightly build. same if user from nightly sends to 0.13.0 release. For both cases when app is on background ios/Android and when app is not running on Android. 

#### Expected behavior

[comment]: # (Describe what you expected to happen.)
push notification is shown if message was sent in 1:1 chat
#### Actual behavior
no push notification is shown
[comment]: # (Describe what actually happened.)

### Reproduction

[comment]: # (Describe how we can replicate the bug step by step.)

- user1 installed nightly and user2 installed 0.13.0 release
- 2 users add each other to contacts, both have password saved
- user1 moves app to background
- user2 sends text msg in 1:1 chat with user1
- check if user1 gets push notification

### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version:  0.14.0 nightly from july 10 and 0.13.0 release
- Operating System: Android 6,9 and iOS 12.3

### LINKED ISSUE #8742: [F-Droid] Make Firebase optional or remove it
## Problem

In the process of researching the build process for F-Droid in #8512 it [turned out](https://github.com/status-im/status-react/issues/8512#issuecomment-520702397) that we cannot submit Status to their database if it includes [Firebase](https://en.wikipedia.org/wiki/Firebase) in its dependencies.

From their [Inclusion Policy](https://f-droid.org/en/docs/Inclusion_Policy/):

>We cannot build apps using proprietary tracking/analytic dependencies like Crashlytics and __Firebase__. Please talk to upstream about an untainted build flavor (either using a FLOSS analytics software like ACRA or by removing non-free dependencies completely).

## Implementation

Either:
* Remove use of Firebase entirely
* Create an `fdroid` build flavour that does not contain Firebase

## Acceptance Criteria

Status for Android can be built without any Firebase dependencies included.

## Notes

I'm not sure why we even included Firebase in the first place considering it [includes analytics](https://firebase.google.com/docs/analytics) and we seem to be using them:
https://github.com/status-im/status-react/blob/3b8d8d0013b4b2dd7cbe3b55ba44ea0f5792e29b/android/app/google-services.json#L4
https://github.com/status-im/status-react/blob/3b8d8d0013b4b2dd7cbe3b55ba44ea0f5792e29b/android/app/google-services.json#L27-L29
I thought that went against our [Principles](https://status.im/about/):

>#### IV. Privacy
>Privacy is the power to selectively reveal oneself to the world. For us, it's essential to protect privacy in both communications and transactions, as well as being a pseudo-anonymous platform. Additionally, we strive to provide the right of total anonymity.

### LINKED ISSUE #9136: No Push Notifications when app is closed [Android]
# Bug Report

## Problem
There are no push notifications appear when message is sent to user in 1-1 chat when Status is closed on Android.
No issues when app on background. 

#### Expected behavior

PN with hashed username appears when messages is sent to user in 1-1 chat when Status is closed 

#### Actual behavior
No PN with hashed username appears when messages is sent to user in 1-1 chat when Status is closed 

#### Acceptance Criteria
On Android, when Status is closed and message is sent in 1-1 chat to this user by the contact who is in the user contact list, - PN appears with the hashed username (because db is closed at this moment)

### Reproduction

- Open Status on two devices and create UserA and UserB
- UserA adds UserB in contacts 
- UserA closes Status app
- UserB sends test message to UserA
- UserA checks whether PN appears

### Additional Information
- Status version: develop  `0.14.0 (2019100703)`

- Operating System: this issue is about Android only (for iOS it's separate issue)

#### Logs

logcat when message is received while Status is closed
```
10-07 17:58:32.997 16528 16626 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :notifications.callback/on-message
10-07 17:58:33.001 16528 16626 W ReactNativeJS: WARN [status-im.notifications.core:109] - failed to lookup contact from hash, not logged in
10-07 17:58:33.005 16528 16626 W ReactNativeJS: WARN [status-im.notifications.core:109] - failed to lookup contact from hash, not logged in
10-07 17:58:33.009 16528 16626 D ReactNativeJS: DEBUG [status-im.notifications.core:226] - handle-on-message app-state: active view-id: nil current-chat-id: nil from: 0x4d6ec88b force: true
10-07 17:58:33.017 16528 16626 D ReactNativeJS: DEBUG [status-im.notifications.background:21] - message-handler-fn called {"ttl":2419200,"sentTime":1570460310119,"messageId":"0:1570460310123549%2bc945a9f9fd7ecd","from":"854811651919","data":{"msg-v2":"{\"from\":\"0x4d6ec88b\",\"to\":\"0x62102113\",\"id\":\"0xcc05a289\"}"}}
10-07 17:58:33.021 16528 16626 D ReactNativeJS: DEBUG [status-im.notifications.background:24] - dispatching :notifications.callback/on-message to display background message {:from "0x4d6ec88b", :to "0x62102113", :id "0xcc05a289"}
```

### LINKED ISSUE #9179: Temporarily delete PN across the app
# Problem
Currently, we have a bunch of issues related to PN and enabling notifications.
Basically PNs are not working at all when app is closed and some [other issues ](https://github.com/status-im/status-react/issues?q=is%3Aissue+label%3Apush-notifications+is%3Aopen)
Also to have permission to send PNs "Enable notifications" screen was introduced. There are several issues with it currently:

1) User can't enable notifications for the app if he restores his account from seed phrase (IOS)
Literally. There is no system option. The same behaviour if user chooses "May be later" - there is no option to enable notifications later.
For Android it is enabled by default, but screen "Enable notifications" is not shown during onboarding, so the user is not aware of it.

2) System notifications are always enabling on Android (regardless of user choice)
So if the user chooses "Maybe later" - it is enabled anyway

The only flow which is working right now (it is fixed in new onboarding PR) - if user enables notifications from this screen.

In order to avoid all these issues PNs should be temporary disabled for V1 (confirmed by @rachelhamlin )

## Implementation
To delete PNs:

* [ ] 1. disable notifications in app and system popups with permissions to send notifications
* [ ] 2. remove screen "Enable notifications" from onboarding
* [ ] 3. remove "Notifications" from Profile

### PR DESCRIPTION
**Pull request by yenda**
fix #9179
fix #9136 (no notification no problem)
fix #8742 (firebase removed)
fix #8555 notifications removed
fix #8554 no more notifications
fix #8550 fix #8217 fix #8216
fix #7438 ios doesn't deserve notifications anyway

Remove Firebase, Google Services and all the Google ads crap from the project.
Remove fcm-token and device infos
Remove push-notifications

# Test protocol

- install app on different devices (iPhone 6 plus, OP7pro)
- create account, recover account, pair devices
- receive 1-1 messages that would normally trigger notifications

What could go wrong:
- UI element related to notifications still present (currently removed: enable notification screen and profile notification setting item)
- leftover notification code could crash the app

### CODE REVIEW

File: src/status_im/pairing/core.cljs [hunk 1]
```diff
@@ -66,11 +66,10 @@
   (sort (partial compare-installation our-installation-id) installations))
 
 (defn pair-installation [cofx]
-  (let [fcm-token         (get-in cofx [:db :notifications :fcm-token])
-        installation-id (get-in cofx [:db :multiaccount :installation-id])
+  (let [installation-id (get-in cofx [:db :multiaccount :installation-id])
         installation-name (get-in cofx [:db :pairing/installations installation-id :name])
         device-type     utils.platform/os]
-    (protocol/send (transport.pairing/PairInstallation. installation-id device-type installation-name fcm-token) nil cofx)))
```
Comment (yenda) [inline src/status_im/pairing/core.cljs:73]: @cammellos what needs to be done on go side to remove fcm tokens and device infos? I think there is also some firebase stuff on go side?

**Comment by yenda** [comment]
@Serhy @churik can you check the issues I marked as resolved and tell me if you agree to close them with this PR? I will test the app myself today, you can also do additional testing if you want.

**Comment by churik** [comment]
@yenda ofc.
let's start from e2e then

**Comment by yenda** [comment]
@churik I tested all the steps described in the summary on both devices.

File: src/status_im/contact/core.cljs [hunk 2]
```diff
@@ -4,7 +4,7 @@
    [status-im.multiaccounts.model :as multiaccounts.model]
    [status-im.transport.filters.core :as transport.filters]
    [status-im.contact.db :as contact.db]
-   [status-im.contact.device-info :as device-info]
+
```
Comment (cammellos) [inline src/status_im/contact/core.cljs]: extra line?

File: src/status_im/pairing/core.cljs [hunk 3]
```diff
@@ -4,7 +4,7 @@
             [status-im.chat.models :as models.chat]
             [status-im.contact.core :as contact]
             [status-im.contact.db :as contact.db]
-            [status-im.contact.device-info :as device-info]
+
```
Comment (cammellos) [inline src/status_im/pairing/core.cljs]: extra line?

**Review by cammellos — APPROVED** [review]

File: src/status_im/contact/core.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/contact/core.cljs]: I naively though that cljfmt would remove them

**Comment by yenda** [comment]
@churik do you want to run some additional manual tests before this is merged?

**Comment by churik** [comment]
Tested on IOS 13.0 and Android 8.0:
- no PNs - no issues mentioned in description;

Additionally checked:
- save password
- reopening app
- fetching messages from offline
- adding / removing from contacts
- creating account with keycard.

Ready to go from my side.
Thx for your work @yenda! 
cc @rachelhamlin - no more PNs for now.

**Comment by rachelhamlin** [comment]
What happens if we decide to revive (suboptimal, centralized) PNs down the line at the request of iOS users? Is it prudent to remove Firebase entirely right now @yenda @PombeirP?

**Comment by yenda** [comment]
@rachelhamlin we can then in the worst case simply revert the commit. But reintroducing Firebase would be a bad decision.

**Comment by ghost** [comment]
`mobile/js_files/yarn.lock` changed. Pinging @jakubgs, @PombeirP, and @corpetty

**Pull request merged**